crates/runtime/src/traphandlers/unix.rs

```diff
@@ -191,19 +200,34 @@ unsafe fn get_pc(cx: *mut libc::c_void, _signum: libc::c_int) -> *const u8 {
                 _ => 0,
             };
             let cx = &*(cx as *const libc::ucontext_t);
-            (cx.uc_mcontext.psw.addr - trap_offset) as *const u8
+            (
+                (cx.uc_mcontext.psw.addr - trap_offset) as *const u8,
+                *(cx.uc_mcontext.gregs[15] as *const usize),
+            )
         } else if #[cfg(all(target_os = "macos", target_arch = "x86_64"))] {
             let cx = &*(cx as *const libc::ucontext_t);
-            (*cx.uc_mcontext).__ss.__rip as *const u8
+            (
+                (*cx.uc_mcontext).__ss.__rip as *const u8,
+                (*cx.uc_mcontext).__ss.__rbp as usize,
+            )
         } else if #[cfg(all(target_os = "macos", target_arch = "x86"))] {
             let cx = &*(cx as *const libc::ucontext_t);
-            (*cx.uc_mcontext).__ss.__eip as *const u8
+            (
+                (*cx.uc_mcontext).__ss.__eip as *const u8,
+                (*cx.uc_mcontext).__ss.__ebp as usize,
+            )
         } else if #[cfg(all(target_os = "macos", target_arch = "aarch64"))] {
             let cx = &*(cx as *const libc::ucontext_t);
-            (*cx.uc_mcontext).__ss.__pc as *const u8
+            (
+                (*cx.uc_mcontext).__ss.__pc as *const u8,
+                (*cx.uc_mcontext).__ss.__opaque_fp as usize,
```

Comment:
This was marked as resolved, but inside the resolved comment is an indication that this isn't the right name?

Locally I get:

```
$ AR=true CC=true cargo check --target aarch64-apple-darwin -p wasmtime-runtime --features posix-signals-on-macos
    Checking wasmtime-runtime v0.40.0 (/home/acrichto/code/wasmtime/crates/runtime)
error[E0609]: no field `__opaque_fp` on type `__darwin_arm_thread_state64`
   --> crates/runtime/src/traphandlers/unix.rs:223:40
    |
223 |                 (*cx.uc_mcontext).__ss.__opaque_fp as usize,
    |                                        ^^^^^^^^^^^ unknown field
    |
    = note: available fields are: `__x`, `__fp`, `__lr`, `__sp`, `__pc` ... and 2 others

For more information about this error, try `rustc --explain E0609`.
error: could not compile `wasmtime-runtime` due to previous error
```

Reply:
Ah seems something got lost in a rebase.